src/reducers/MapReducer.js

```diff
@@ -75,6 +150,14 @@ export default (state = INITIAL_STATE, action) => {
         case FACEBOOK_LOGOUT_COMPLETE:
             return INITIAL_STATE;
 
+        case FILTER_CRITERIA_UPDATED:
+            const {filterKey} = payload;
+            const updatedFilterKeys = updateFilterKeys(state, filterKey);
+            const newlyFilteredSites = filterSites(state, updatedFilterKeys);
+
+            return {...state, filterCriteriaKeys: updatedFilterKeys, displaySites: newlyFilteredSites};
```

Comment:
Whenever we have a new filter value, we need to add it to the ongoing list, and also filter all sites and return them as the new `displaySites`.